File: DonatorSange-Angular/src/app/models/donor.ts
import { Address } from "./address"
import { DonorData } from "./donorData";
import { User } from "./user";
import { DonationRequest } from "./donationRequest";
import { Gender } from "./gender";

export class Donor{
    id:number;
    address:Address;
    phone:string;
    email:string;
    gender:Gender;
    donorData:DonorData;
    donationRequest:DonationRequest;
    user:User;
    
    static initialize(){
        let donor = new Donor();
        donor.id = 0;
        donor.address = Address.initialize();
        donor.donationRequest = DonationRequest.initialize();
        donor.donorData = null;
        donor.email = null;
        donor.gender = Gender.initialize();
        donor.phone = null;
        donor.user = new User(0, null, null, null, null, null);
        return donor;

    }
}

File: AntiClimacus/DataAccessLayer/Entities/Donor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DataAccessLayer.Entities
{
    public class Donor
    {
        public int Id { get; set; }
        public int AddressId { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public int GenderId { get; set; }
        public int UserId { get; set; }
        public bool HasCondition { get; set; }

        public Gender Gender { get; set; }
        public User User { get; set; }
        public DonorData DonorData { get; set; }
        public Address Address { get; set; }
        public ICollection<DonationRequest> DonationRequests { get; set; }
    }
}
